lib/compile.js

```diff
@@ -105,6 +106,26 @@ function compileLoad(loader, load, compileOpts, cache) {
     if (format == 'cjs')
       mappedLoad.source = mappedLoad.source.replace(hashBangRegEx, '');
     return Promise.resolve()
+    .then(function () {
+      var metadata = mappedLoad.metadata;
+      if (metadata.sourceMap)
+        return;
+      var sourceMap = sourceMappingURL.getFrom(mappedLoad.source);
+      if (!sourceMap)
+        return;
+      mappedLoad.source = sourceMappingURL.removeFrom(mappedLoad.source);
+      if (sourceMap.startsWith('data:')) {
+        // inline source map
+        metadata.sourceMap = JSON.parse(dataUriToBuffer(sourceMap).toString('utf8'));
```

Comment:
Unfortunately we don't have a good warning system at the moment, and yes we don't want to just be writing to the console from within the API.

I was thinking of just having a silent error, because source maps fail silently in the browser when they don't work so that seemed like it may be an ok precedent. But let me know if that sounds unreasonable?